### PR TITLE
Improve Github alerter for Mako

### PR DESCRIPTION
**Pull request by chizhg**
Improve mako alerter:
1. Edit the old comment if there is an existed Github issue
2. Code clean up

### CODE REVIEW

File: test/mako/alerter/github/issue.go [hunk 1]
```diff
@@ -87,65 +88,72 @@ func Setup(org, repo, githubTokenPath string, dryrun bool) (*IssueHandler, error
...
-		}
-		comment := fmt.Sprintf(newIssueCommentTemplate, desc)
-		if err := gih.addComment(org, repo, *issue.Number, comment, dryrun); err != nil {
-			return err
-		}
-		// If one issue with the same title has been closed, reopen it and add new comment
-	} else if *issue.State == string(ghutil.IssueCloseState) {
-		if err := gih.reopenIssue(org, repo, *issue.Number, dryrun); err != nil {
-			return err
+			return fmt.Errorf("failed to create a new issue for test %q", testName)
```
Comment (vagababov) [inline test/mako/alerter/github/issue.go]:
```suggestion
			return fmt.Errorf("failed to create a new issue for test %q: %w", testName, err)
```

File: test/mako/alerter/github/issue.go [hunk 2]
```diff
@@ -87,65 +88,72 @@ func Setup(org, repo, githubTokenPath string, dryrun bool) (*IssueHandler, error
...
 		}
-		comment := fmt.Sprintf(reopenIssueCommentTemplate, desc)
-		if err := gih.addComment(org, repo, *issue.Number, comment, dryrun); err != nil {
-			return err
+		commentBody = fmt.Sprintf(issueSummaryCommentTemplate, desc)
+		if err := gih.addComment(*issue.Number, commentBody); err != nil {
+			return fmt.Errorf("failed to add comment for new issue %d", *issue.Number)
```
Comment (vagababov) [inline test/mako/alerter/github/issue.go]:
```suggestion
			return fmt.Errorf("failed to add comment for new issue %d: %w", *issue.Number, err)
```

File: test/mako/alerter/github/issue.go [hunk 3]
```diff
@@ -87,65 +88,72 @@ func Setup(org, repo, githubTokenPath string, dryrun bool) (*IssueHandler, error
...
 		}
 	} else {
```
Comment (vagababov) [inline test/mako/alerter/github/issue.go]: perhaps, `return nil` and unindent here?

File: test/mako/alerter/github/issue.go [hunk 4]
```diff
@@ -87,65 +88,72 @@ func Setup(org, repo, githubTokenPath string, dryrun bool) (*IssueHandler, error
...
-				return err
+		issueNumber := *issue.Number
+		// If the issue has been closed, reopen it
+		if *issue.State == string(ghutil.IssueCloseState) {
+			if err := gih.reopenIssue(issueNumber); err != nil {
+				return fmt.Errorf("failed to reopen issue %d", issueNumber)
+			}
+			commentBody := fmt.Sprintf(reopenIssueCommentTemplate, desc)
+			if err := gih.addComment(issueNumber, commentBody); err != nil {
+				return fmt.Errorf("failed to add comment for reopened issue %d", issueNumber)
```
Comment (vagababov) [inline test/mako/alerter/github/issue.go]:
```suggestion
				return fmt.Errorf("failed to add comment for reopened issue %d: %w", issueNumber, err)
```

Comment (vagababov) [reply test/mako/alerter/github/issue.go]: same below

File: test/mako/alerter/github/issue.go (hunk 3 above)
Comment (chizhg) [reply test/mako/alerter/github/issue.go]: Done.

**Comment by vagababov** [comment]
Still WIP?

**Comment by chizhg** [comment]
> Unfortunately it seems that `pkg` has not yet migrated to go1.13, so for now, can you make al %w -> %v, we can bulk convert them later.
> Otherwise looks good.

Yeah I saw the discussions, reverted them.
Now this PR is ready.

/cc @adrcunha

**Pull request merged**